File: python/greatest-num.py
# Python program that finds the greatest of n numbers

try:
    count = int(input("Number count: "))

except ValueError:
    print("Invalid count")
    quit()

num_list = []

for i in range(count):
    try:
        num_list.append(float(input(f"Type a number: ")))

    except ValueError:
        print("Invalid number")

print(f"Greatest value is: {max(num_list)}")

File: ts/greatestNum.ts
const greatestNum = (n: Array<number>) => {
  let greatest = n[0];

  n.forEach((number) => {
    if (number > greatest) greatest = number;
  });

  return greatest;
};

console.log(greatestNum([0]));
console.log(greatestNum([0, 1, 2, 3]));
console.log(greatestNum([4, 1.5, 222, 30]));
